### PR TITLE
Allow clients to send custom tags to Zipkin server

### PR DESCRIPTION
**Pull request by pranshuparmar**
Provide clients with capability to send custom tags to Zipkin server.
Client needs to add a map of tags to exchange property camel.custom.tags and camel-zipkin will automatically add these tags to Zipkin span.

### CODE REVIEW

**Comment by oscerd** [comment]
Since @davsclaus was the original author of the component, I'll wait for him on reviewing the PR

File: components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java [hunk 1]
```diff
@@ -56,5 +56,15 @@ void onRequest(Exchange exchange, SpanCustomizer span) {
                 cache.reset();
             }
         }
+        
+        Object clientTagsObj = exchange.getProperty("camel.custom.tags");
```
Comment (davsclaus) [inline components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java]: Cant we use `camel.client.customtags` eg with `camel.client` as prefix.

And then you can use `exchange.getProperty("xxx", Map.class);

File: components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java [hunk 2]
```diff
@@ -56,5 +56,15 @@ void onRequest(Exchange exchange, SpanCustomizer span) {
...
+        if (clientTagsObj != null) {
+        	Map<String, String> clientTags = (Map)clientTagsObj;
+	        if (clientTags != null && !clientTags.isEmpty()) {
+	        	for (Map.Entry<String, String> tag: clientTags.entrySet()) {
+	        		span.tag("custom." + tag.getKey(), tag.getValue());
```
Comment (davsclaus) [inline components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java]: Fix checkstyle, eg run `mvn compile -P sourcecheck` and fix the errors it report

**Comment by oscerd** [comment]
@pranshuparmar can you please update the PR?

File: components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java (hunk 1 above)
Comment (pranshuparmar) [reply components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java]: Have updated property name to camel.client.customtags and used getProperty() method with class type.

File: components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java (hunk 2 above)
Comment (pranshuparmar) [reply components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinClientRequestAdapter.java]: Fixed checkstyle issues and added unit test.

**Pull request merged**